src/main/java/com/entropyteam/entropay/employees/services/ReportService.java

```diff
@@ -121,9 +128,48 @@ public Page<EmployeeReportDto> getEmployeesReport(ReactAdminParams params) {
     }
 
     public List<Employee> getFilteredEmployeesList(Filter filter) {
-        if(filter.getGetByFieldsFilter().containsKey(ACTIVE_CONTRACT)) {
+        if (filter.getGetByFieldsFilter().containsKey(ACTIVE_CONTRACT)) {
             return employeeRepository.getEmployeesWithAtLeastAnActiveContract();
         }
         return employeeRepository.findAllByDeletedIsFalseAndActiveIsTrue();
     }
+
+    public Page<PtoReportDto> getPtosByEmployeesReport(ReactAdminParams params) {
+        List<Employee> employeeList = employeeRepository.findAllByDeletedIsFalse();
+
+        Map<UUID, List<Assignment>> employeeAssignmentsMap = assignmentRepository.findAllByDeletedIsFalse()
+                .stream()
+                .collect(Collectors.groupingBy(a -> a.getEmployee().getId()));
```

Comment:
this way we are getting all employess + all asignments in the DB. Let's gets only those employees that has a pto in ptoList